File: src/jionchu/Week2/예상 대진표.kt
/* Programmers
 * 2017 팁스타운 - 예상 대진표
 * created on 2021.09.27
 * created by jionchu */

class Solution {
    fun solution(n: Int, a: Int, b: Int): Int {
        var answer = 0
        var a = a-1
        var b = b-1
        
        while(a != b) {
            answer++
            a/=2
            b/=2
        }

        return answer
    }
}

File: src/dohyun/2week/예상 대진표.js
function solution(n, a, b) {
  for (let i = 1; ; i++) {
    a = a % 2 === 0 ? a / 2 : parseInt(a / 2) + 1;
    b = b % 2 === 0 ? b / 2 : parseInt(b / 2) + 1;
    if (a === b) return i;
  }
}
